File: 1005-UnivaluedBinaryTree/1005-UnivaluedBinaryTree.java
// Last updated: 8/5/2025, 2:56:13 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isUnivalTree(TreeNode root) {
        if (root == null)
            return true;
        return ((root.left == null || root.left.val == root.val && isUnivalTree(root.left)) && 
                (root.right== null || root.right.val== root.val && isUnivalTree(root.right)));
    }
}

File: 1005-UnivaluedBinaryTree/1005-UnivaluedBinaryTree.py
# Last updated: 8/5/2025, 2:56:10 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isUnivalTree(self, root: TreeNode) -> bool:
        store = set()
        def traverse(root):
            if root:
                store.add(root.val)
                traverse(root.left)
                traverse(root.right)
        traverse(root)
        if len(store) <= 1:
            return True
        else:
            return False